initfiles/bin/init_thorslave

```diff
@@ -78,42 +81,33 @@ start_slaves()
     ulimit -Sn hard > /dev/null 2>&1
     [[ $? -ne 0 ]] && log "Failed to set ulimit for number of file descriptors open"
 
-    log "slave(${ip}) init"
+    log "slave init"
     log "slave(s) starting"
 
     # create symlink for easier identification of slaves by compName
     ln -s -f $deploydir/thorslave_lcr ${slavename}
 
-    # sync to current master thorgroup
-    log "rsync -e ssh -o LogLevel=QUIET -o StrictHostKeyChecking=no ${master}:${instancedir}/thorgroup ${instancedir}/thorgroup.slave"
-    rsync -e "ssh -o LogLevel=QUIET -o StrictHostKeyChecking=no" $master:$instancedir/thorgroup $instancedir/thorgroup.slave
-
-    let "slavenum = 1";
-    for slave in $(cat $instancedir/thorgroup.slave); do
-        slaveip=${slave/:*/}
-        if [[ ${slaveip} = ${ip} ]]; then
-            slaveport=${slave/*:/}
-            if [[ "$slaveport" = "" ]]; then
-                slaveport=$THORSLAVEPORT
-            fi
-            log "$slavename  master=$master:$masterport slave=.:$slaveport slavenum=$slavenum logDir=$logpth/$hpcc_compname"
-            ./$slavename master=$master:$masterport slave=.:$slaveport slavenum=$slavenum logDir=$logpth/$hpcc_compname 2>/dev/null 1>/dev/null &
-            slavepid=$!
-            if [[ "$slavepid" -eq "0" ]]; then
-                log "failed to start"
-            else
-                log "slave pid $slavepid started"
-                PID_NAME="$PID/${slavename}_${slavenum}.pid"
-                echo $slavepid > $PID_NAME
-            fi
+    slavenum=${processnum}
+    for (( slave=1; slave<=${slavespernode}; slave++ ))
+    do
+        log "$slavename master=$master:$masterport slave=.:$slaveport slavenum=$slavenum logDir=$logpth/$hpcc_compname"
+        ./$slavename master=$master:$masterport slave=.:$slaveport slavenum=$slavenum logDir=$logpth/$hpcc_compname 2>/dev/null 1>/dev/null &
+        slavepid=$!
+        if [[ "$slavepid" -eq "0" ]]; then
+            log "failed to start"
+        else
+            log "slave pid $slavepid started"
+            PID_NAME="$PID/${slavename}_${slavenum}.pid"
+            echo $slavepid > $PID_NAME
         fi
-        let "slavenum += 1";
+        slavenum=$((${slavenum} + ${slavespernode}))
```

Comment:
Pretty sure this should be something like 'slavenum=$((${slavenum} + ${number_of_nodes}))'
Slavespernode should be used for how many iterations of the for loop we do (to set up the number of instances of thorslaves) but the actual offset for slavenum should come from the number of nodes we have.

If you preflight the mythor cluster, setup with 3 thorslave nodes, with 2 slavesPerNode and 1 channelsPerSlave.  You'll see an incorrect return on the slavenum to ip in the preflight.